mono/utils/mono-tls.h

```diff
@@ -85,11 +85,11 @@ gint32 mono_tls_get_tls_offset (MonoTlsKey key);
 gpointer mono_tls_get_tls_getter (MonoTlsKey key, gboolean name);
 gpointer mono_tls_get_tls_setter (MonoTlsKey key, gboolean name);
 
-gpointer mono_tls_get_thread (void);
-gpointer mono_tls_get_jit_tls (void);
-gpointer mono_tls_get_domain (void);
-gpointer mono_tls_get_sgen_thread_info (void);
-gpointer mono_tls_get_lmf_addr (void);
+struct _MonoInternalThread *mono_tls_get_thread (void);
```

Comment:
This is exposing structs from other subsystems here.
The casts should be on the use site instead.

Reply:
It doesn’t expose what they have, just that they exist & their name. They remain opaque. You can now see them more easily in debugger.